docs/adr/adr_0012.md

```diff
@@ -0,0 +1,129 @@
+
+When deploying cluster scans types you have to make sure that either:
+
+1. All referenced ConfigMaps / Secrets exist in all namespaces. This will likely be done by a script / operator generating these configmaps and persisting them into every namespace
+2. Remove all dependencies to ConfigMaps / Secrets from your scanTypes and move these either into the container image of the scanner, or by specifing a initContainer in the ScanType which copies these files over to volume shared with the scanner container. Some more references and discussions on ways to improve this in the future can be found in [ADR-0009: Architecture for pre-populating the file system of scanners"](./adr-0009)
+
+### `ClusterParseDefinition`
+
+Other then being cluster scoped `ClusterParseDefinition`s are identical to the existing `ParseDefinition` CRD.
+
+When a scan job has completed and the results need to be parsed, the operator should first look for `ParseDefinition`s in the `Scan`s namespace matching the `scantype.spec.extractResults.type` / `clusterscantype.spec.extractResults.type` name configured in the ScanType.
+If that is not found the operator should look for `ClusterParseDefinition`s with the same name.
+No matter which was picked the `Job` for the parser always has to be started in the namespace of the `Scan`.
+
+### `ClusterCascadingRule`
+
+Other then being cluster scoped `ClusterCascadingRule`s are identical to the existing `CascadingRule` CRD.
+
+When the cascading scans hook starts, the hook should fetch the list of both `CascadingRule`s and `ClusterCascadingRule`s and merge them together into one list which then get's evaluated against the scan just like it is now. All scans started from it have to be in the same namespace of the original `Scan`.
+
+#### Deduplication of CascadingRules
+
+One potential issue with CascadingRule and ClusterCascadingRule is that in contrast to ScanType and ParseDefinition both namespaced and cluster wide types are used. This can be problematic when both the namespace as well as the cluster level have similar / equal CascadingRule installed as this could lead to both producing the same scan which would then be executed twice.
+
+Potential workarounds here would be to introduce an additional field to the CascadingRule spec which allows us to define a sort of "deduplication" / replacement mechanism.
+
+### `ClusterScanCompletionHook`
+
+`ClusterScanCompletionHook` behave mostly like their namespaced counterpart, when the parser completes the operator fetches the list of ScanCompletionHooks from Scans namespace (respecting the hookSelector of the Scan) and the `ClusterScanCompletionHook`s (also respecting the hookSelector of the Scan (this might have to be discussed...)) are merged and ordered according to the respective priorities set on them.
```

Comment:
```suggestion
`ClusterScanCompletionHook` behave mostly like their namespaced counterpart, when the parser completes, the operator fetches the list of ScanCompletionHooks from Scans namespace (respecting the hookSelector of the Scan) and the `ClusterScanCompletionHook`s (also respecting the hookSelector of the Scan (this might have to be discussed...)) are merged and ordered according to the respective priorities set on them.
```